metaspace2020/metaspace
# Feat/tic ds comparison and browser page

### Description

Add TIC feature to ds comparison page and imzml browser page and add optical image opacity to ds comparison page. 
#937

### How to test it

* Add the tic feature flag to test it (feat=tic).
* As the imzml browser backend routes are a POC, in order to test the front-end, you need to run the backend as a standalone server. To test it, follow these steps:

1 - Run the metaspace project
2 - Go to the the folder metaspace/metaspace/browser and register the sm dev package:  python3 setup.py develop
3 - Go to the folder /metaspace/metaspace/browser/sm/browser and run the server: uvicorn api:app --reload 
4 - Select the inputPath of the dataset you want to check the functionality, add its value to the attribute s3_path and pre-process it (via postman):
     route: http://127.0.0.1:8000/preprocess
     body: {
             "s3_path": "s3://sm-engine-dev/ddd7f0e5-ff74-4f61-96d4-60f40dfb4c3c",
     }
     * This step is needed, because I still didn't create a back-end route to check if the temp files generated already exists (i didnt do because I dont know how this will be in the production env).
5 - After pre-process the dataset, you can access the imzml browser page on the front-end: http://localhost:8999/dataset/<dataset_id>/browser    

#### Changes

##### Webapp

###### MainImage
- [x] Added normalization props

###### MainImageHeader
- [x] Added show normalization badge props

###### DatasetComparisonPage
- [x] Added  tic normalization

###### DatasetComparisonGrid
- [x] Added normalization
- [x] Added optical Image visibility control

###### DatasetBrowserPage
- [x] Added normalization

#### Tests
 
##### Front end
 
 
###### Browsers and resolutions


![image](https://user-images.githubusercontent.com/35172605/134609696-89504584-f28a-422e-8165-201e552a16f2.png)
![image](https://user-images.githubusercontent.com/35172605/134609846-307fe2b3-8443-42dc-9af4-90ec39af7c2c.png)
